docs/editing-components.md

```diff
@@ -0,0 +1,376 @@
+   some other block (the `Hyperlink` atom block, in this case).
+5. The `Meta` class defines some properties on the `RelatedContent` block
+   that are used by the Wagtail admin or in rendering the block.
+6. The `icon` property tells Wagtail what icon to use in the editor
+   for the button you use to add a `RelatedContent` block to a StreamField.
+   Icon options can be found in the Wagtail style guide when running locally:
+   http://localhost:8000/admin/styleguide/#icons
+7. The optional `label` property overrides the text of that same button;
+   if `label` is not set, Wagtail will generate one from the name of the block.
+8. The `template` property is a pointer to the HTML template used to render
+   this component. [See below for more on templates.](#the-html-template)
+
+
+### Adding it to a StreamField
+
+Components are made available in the page editing interface
+by adding them to one of a page types's StreamFields.
+These are usually the first things in a page's class definition.
+For example, see this snippet from `blog_page.py`:
+
+```python
+class BlogPage(AbstractFilterPage):
+    content = StreamField([
+        ('full_width_text', organisms.FullWidthText()),
+        ('info_unit_group', organisms.InfoUnitGroup()),
+        ('expandable', organisms.Expandable()),
+        ('well', organisms.Well()),
+        ('email_signup', organisms.EmailSignUp()),
+        ('feedback', v1_blocks.Feedback()),
+        ('image_text_50_50_group', organisms.ImageText5050Group()),
+    ])
+    …
+```
+
+This sets up a StreamField named `content` that allows for
+the insertion of any of those seven listed blocks into it.
+
+To make the `RelatedContent` module (shown above) available to this StreamField,
+we'd add a new entry to this list following the same format:
+`('related_content', molecules.RelatedContent()),`.
+
+Most page types have a couple StreamFields (`header` and `content`) in the
+general content area (the first tab on an editing screen), and most also share
+[a common `sidefoot` StreamField](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/models/base.py#L95-L107)
+(so named for the fact that it appears on the right side on some page types,
+but in the footer on others) on the sidebar tab.
+
+
+---
+
+!!! note "Warning: Migrations may be required!"
+    Changes to either of the Python bits will usually require
+    a new schema migration file, and _may_ require a data migration.
+    See
+    ["Creating migrations for StreamField blocks"](#creating-migrations-for-streamfield-blocks)
+    below.
+
+With the Python pieces covered, it's time to talk about the front end.
+First: the markup.
+
+
+### The HTML template
+
+As mentioned above, each module's Python class has a `Meta` class
+that defines the location of its template file.
+In the `RelatedContent` example,
+it was `template = '_includes/molecules/related-content.html'`.
+That path is relative to `cfgov/jinja2/v1/`.
```

Comment:
This is true in this particular case, but other block template HTML is loaded elsewhere. This statement should be made more general to refer to how Django template paths are used.

Reply:
Agreed. See above about blocks that will only be used inside a certain app. 

We also have a bad habit of putting absolutely all templates regardless of which Django app uses them into this one template path. That makes the packaging problem much harder.

Reply:
@chosak Not sure I understand enough about general Django template path behavior to address this. Can you suggest something?

Reply:
Sure, how about this:

Frontend rendering of Wagtail StreamField blocks to HTML can be controlled by writing a Django template and associating it with the block type. A custom block definition can specify the template path in its `Meta` class:

```py
from wagtail.core import blocks

class PersonBlock(blocks.StructBlock):
    name = blocks.CharBlock()
    email = blocks.EmailBlock()

    class Meta:
        template = 'myapp/blocks/person_block.html'
```

StreamField block templates are loaded by the Django template loader in the same way that Django page templates are. The specified template path must be loadable by one of the Django template engines configured in [`settings.TEMPLATES`](https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-TEMPLATES) (for this project, the [Django templates backend](https://docs.djangoproject.com/en/2.1/topics/templates/#django.template.backends.django.DjangoTemplates) and the [Jinja2 backend](https://docs.djangoproject.com/en/2.1/topics/templates/#django.template.backends.jinja2.Jinja2)). See [the Django documentation](https://docs.djangoproject.com/en/1.11/topics/templates/#usage) for more details on the search algorithm used to locate a template.

Reply:
Added in 968e783c4997ad98d51c1f8310159a1a68772b07